frontend/src/Student/components/OtherMediaSubmissionForm.js

```diff
@@ -0,0 +1,400 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Form, FormGroup, FormFeedback, Label, Button, Row, Col } from 'reactstrap'
+import styled from 'styled-components'
+import Dropzone from 'react-dropzone'
+import { Formik, Field } from 'formik'
+import yup from 'yup'
+
+import SuccessModal from './SuccessModal'
+
+const Header = styled.h1`
+  margin-bottom: 10px;
+`
+
+const SubHeader = styled.h3`
+  margin-bottom: 25px;
+`
+
+const PreviewImage = styled.img`
+  height: 100%;
+`
+
+const ButtonContainer = styled.div`
+  margin-top: 50px;
+`
+
+class OtherSubmissionForm extends Component {
+  static propTypes = {
+    user: PropTypes.shape({
+      username: PropTypes.string
+    }).isRequired,
+    forShow: PropTypes.shape({
+      id: PropTypes.number,
+      name: PropTypes.string
+    }).isRequired,
+    handleImageUpload: PropTypes.func.isRequired,
+    handlePDFUpload: PropTypes.func.isRequired,
+    previewFile: PropTypes.object.isRequired,
+    create: PropTypes.func.isRequired,
+    done: PropTypes.func.isRequired
+  }
+
+  static defaultProps = {
+    previewFile: {}
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      showModal: false
+    }
+  }
+
+  renderFileUpload = (field, form) => {
+    const { name } = field
+    const { setFieldValue } = form
+    const {
+      handleImageUpload,
+      handlePDFUpload,
+      previewFile
+    } = this.props
+
+    return (
+      <Dropzone
+        name={name}
+        accept='application/pdf,image/jpeg'
+        style={{
+          alignItems: 'center',
+          cursor: 'pointer',
+          display: 'flex',
+          height: '250px',
+          justifyContent: 'center',
+          textAlign: 'center'
+        }}
+        activeStyle={{
+          borderColor: '#6c6',
+          backgroundColor: '#eee'
+        }}
+        rejectStyle={{
+          borderColor: '#c66',
+          backgroundColor: '#eee'
+        }}
+        className='form-control'
+        onDrop={(acceptedFiles) => {
+          const file = acceptedFiles[0] // Only 1 file per submission
+
+          switch (file.type) {
+            case 'application/pdf':
+              handlePDFUpload(file).then(() => {
+                // Need to use 'this.props' here to get the most up-to-date value – 'previewFile' above will be out-of-date
+                setFieldValue(name, this.props.previewFile.path)
+              })
+              break
+            case 'image/jpeg':
+              handleImageUpload(file).then(() => {
+                // Need to use 'this.props' here to get the most up-to-date value – 'previewFile' above will be out-of-date
+                setFieldValue(name, this.props.previewFile.path)
+              })
+              break
+            default:
+              console.error(`Unknown File Type: ${file.type}`)
```

Comment:
So, this code should _technically_ never be reached. We only `accept='application/pdf,image/jpeg'`, so there would need to be issues w/ the Dropzone or someone would have to be modifying our internals through dev tools.

Reply:
I remember I also added this case because in `react-dropzone`'s docs, they mention that MIME type determination isn't reliable across platforms – if and end user is having issues uploading a file, someone would be able to diagnose the issue by having the end user open their console and see what the error message is.

See the `accept` prop: https://react-dropzone.js.org/